app/uk/gov/hmrc/brm/utils/BrmException.scala

```diff
@@ -37,44 +37,64 @@ trait BRMException extends Controller {
     }
   }
 
-  def notFoundPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+  def notFoundPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream4xxResponse(body, NOT_FOUND, _, _) =>
       logException(method, body, NOT_FOUND)
       Ok(Json.toJson(BirthResponseBuilder.withNoMatch()))
   }
 
-  def forbiddenPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+  def forbiddenPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream4xxResponse(body, FORBIDDEN, _, _) =>
       logException(method, body, FORBIDDEN)
       Ok(Json.toJson(BirthResponseBuilder.withNoMatch()))
   }
 
-  def badRequestPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+  def badRequestPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream4xxResponse(body, BAD_REQUEST, _, _) =>
       logException(method, body, BAD_REQUEST)
       BadRequest
   }
 
-  def badGatewayPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+ def badGatewayPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream5xxResponse(body, BAD_GATEWAY, _) =>
       logException(method, body, BAD_GATEWAY)
       BadGateway
   }
 
-  def gatewayTimeoutPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+  def desConnctionDownPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
+    case e: BadGatewayException if payload.whereBirthRegistered == BirthRegisterCountry.SCOTLAND =>
+      logException(method, s"BadGatewayException: ${e.getMessage}", BAD_GATEWAY)
+      logException(method, s"Response body: ${ErrorResponse.DES_CONNECTION_DOWN}", SERVICE_UNAVAILABLE)
+      ServiceUnavailable(ErrorResponse.DES_CONNECTION_DOWN)
+  }
+
+  def gatewayTimeoutPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream5xxResponse(body, GATEWAY_TIMEOUT, _) =>
       logException(method, body, GATEWAY_TIMEOUT)
       GatewayTimeout
   }
 
-  // TODO implement connection down for GRO
-  def upstreamErrorPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+  def groConnctionDownPF(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
+    case e: Upstream5xxResponse if e.message.contains("GRO_CONNECTION_DOWN") =>
```

Comment:
@manishsw this is returning a 503 if the proxy responds with a InternalServerError 500?